stats/cloud/collector.go

```diff
@@ -204,6 +208,12 @@ func (c *Collector) Run(ctx context.Context) {
 
 		go func() {
 			for {
+				select {
+				case <-c.stopCh:
+					return
+				default:
+				}
+
```

Comment:
Technically a panic possibilty again:
We could already be in the bottom select when an error happens and than the context can be done and we will again push metrics and ... worse close the channel.
I propose that we don't push ... here but only in the bottom for but the aggregation might need to be moved somewhat

Reply:
Maybe adding another select stopCh in the beginning of pushMetrics will be easier and simpler?

Reply:
I dunno ... 
I would prefer if we remove the `pushMetrics` from here and add another channel that is closed here and is listened to in the for below instead of the `context.Done`

Reply:
Not sure I got your point, can you describe more details?

Reply:
My point is that even if everything works correctly we will still pushMetrics twice ... which is IMHO not a good idea ... literally you can still get two different goroutines in pushMetrics (one from this one because the context was `Done` and one from the pushMetrics loop (either cases of the select)) in this case there is nothing stopping them from closing the channel twice ... even with your new select inside of it. 

So I propose that we stick to ... what I think is the design in all of the outputs including this one (but someone at some point deviated from it here) and just have a *single* goroutine pushing metrics and a *single* goroutine aggregating metrics. 

But in order for the *last* metrics to be aggregated before pushed we need to first run the `ctx.Done()` case here and then the one in the for loop that is pushing . To do that I propose that we add another channel local to this function, close it in the `ctx.Done()` branch here and wait for it instead of `ctx.Done()` in the `pushMetric` loop below.

Reply:
@MStoykov make sense, I pushed new patch with testing the number of times we pushed metrics, please take a look.

Reply:
https://github.com/loadimpact/k6/pull/1130/files#diff-11a7e92298be39c5269425095fb75dc9R460